packages/components/pie-button/src/decorators.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * A decorator for specifying a list of valid values for a property.
+ * If this property's setter is called with an invalid value, an error is logged and the default value will be assigned instead.
+ * @param validValues - The array of valid values
+ * @param defaultValue - The value to fall back on
+ * @returns
+ */
+export const validValues = (validValues: any[], defaultValue: any) => {
+    return function (target: any, propertyKey: string) : void {
+        const privatePropertyKey = `_${propertyKey}`;
+
+        Object.defineProperty(target, propertyKey, {
+            get () : any {
+                return target[privatePropertyKey];
+            },
+            set (value: any) : void {
+                const oldValue = target[privatePropertyKey];
+
+                if (!validValues.includes(value)) {
+                    console.error(
+                        `Invalid type value provided: "${value}".`,
```

Comment:
I'd suggest to add more context to the error message, like the component name.

Currently we have a single component, but if we imagine a scenario where the user updates several components to the newest version, it would be nice to know which component is raising the error message without effort.

Reply:
Nice idea, updated.